lib/pages/home_page.dart

```diff
@@ -13,54 +14,131 @@ class HomePage extends StatefulWidget {
 }
 
 class _HomePageState extends State<HomePage> {
-  LocationPermission? gpsPermission;
-  Position? position;
+  LocationPermission? gps = LocationPermission.denied;
+  Position? pos;
 
-  @override
-  void initState() {
-    foo();
+  Future<void> foo() async {
+    gps = await Geolocation.instance.gpsPermission;
+    pos = await Geolocation.instance.position;
   }
 
-  Future<void> foo() async {
-    LocationPermission gps = await Geolocation().turnOnServices();
-    Position pos = await Geolocator.getCurrentPosition(
-        desiredAccuracy: LocationAccuracy.high);
-    setState(() {
-      gpsPermission = gps;
-      position = pos;
-    });
+  Future<void> refreshBtn() async {
+    setState((){});
   }
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        title: const Text('Home'),
-      ),
-      body: SizedBox(
-        height: double.infinity,
-        width: double.infinity,
-        child: Scaffold(
-          body: FlutterMap(
-            options: MapOptions(
-                center: LatLng(position.latitude, position.longitude),
-                zoom: 13.0,
-                minZoom: 2,
-                maxZoom: 18.3,
-                interactiveFlags: InteractiveFlag.all & ~InteractiveFlag.rotate,
-                maxBounds:
-                    LatLngBounds(LatLng(-90, -180.0), LatLng(90.0, 180.0))),
-            children: [
-              TileLayer(
-                urlTemplate:
-                    "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
-                subdomains: const ['a', 'b', 'c'],
+    return FutureBuilder(
+      future: foo(),
+      builder: (context, snapshot) {
+        if (snapshot.connectionState == ConnectionState.waiting) {
+          return Scaffold(
+            appBar: AppBar(),
+            body: Container(
+                width: double.infinity,
+                height: double.infinity,
+                color: Colors.white,
+                child: Column(
+                    crossAxisAlignment: CrossAxisAlignment.center,
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: const [
+                      CircularProgressIndicator(
+                        color: Color(0xff725ac1),
+                        backgroundColor: Colors.white,
+                      )
+                    ])),
+          );
```

Comment:
Toto můžeš značně zjednodušit 😉 
```
return Scaffold(
      appBar: AppBar(),
      backgroundColor: Colors.white,
      body: Center(
        child: CircularProgressIndicator(),
      ),
    );
```
`Center` widget se totiž roztáhne do maximální poskytnuté velikosti a vycentruje ti `child`. Barvu pak můžeš definovat rovnou tomu `Scaffold`, ale celkově na ty barvy by bylo mnohem lepší si nadefinovat nějaký theme. Můžeme se o tom pobavit na meetu ;)